src/cmd/ksh93/tests/wchar.sh

```diff
@@ -21,7 +21,13 @@
 # Tests for \u[...] and \w[...] input and output
 #
 
-locales="en_US.UTF-8 en_US.utf8 en_US.ISO-8859-15 en_US.iso885915 zh_CN.GB18030 zh_CN.gb18030"
+locales="en_US.UTF-8 en_US.utf8 en_US.ISO-8859-15 en_US.iso885915"
+# On FreeBSD 11.1 the native locale subsystem cannot convert the Euro currency symbol from UTF-8 to
+# GB1803. See https://github.com/att/ast/issues/577#issuecomment-396113061.
+if [[ $OS_NAME != "FreeBSD" ]]
+then
+    locales="$locales zh_CN.GB18030 zh_CN.gb18030"
```

Comment:
Are the locales `zh_CN.GB18030` and `zh_CN.gb18030` supported on FreeBSD ? Shall we add a todo note to write separate tests for it ?

Reply:
They are supported but for some reason the U+20AC code point can't be translated. Most likely the FreeBSD GB18030 code table is old and missing that character.

Reply:
@saper Do you think it could be a bug in FreeBSD ?

Reply:
Certainly, will have a look!

Reply:
@saper You might also want to take a look into why the `LC_ALL` env var doesn't supercede the other locale env vars. Create a file with a UTF-8 encoded U+20AC (the Euro currency symbol). Then run `env LC_ALL=C LC_CTYPE=en_US.UTF-8 wc -m ~/x`. It should report `3` and does on all the other platforms (macOS, OpenBSD, GNU Linux). On FreeBSD 11.1 it reports `1`.